src/bpffeature.cpp

```diff
@@ -564,59 +564,91 @@ bool BPFfeature::has_skb_output()
-
-      << std::endl;
-
-  buf << "Kernel features" << std::endl
-      << "  Instruction limit: " << instruction_limit() << std::endl
-      << "  Loop support: " << to_str(has_loop())
-      << "  btf: " << to_str(has_btf())
-      << "  module btf: " << to_str(has_module_btf())
-      << "  Kernel DWARF: " << to_str(has_kernel_dwarf())
-      << "  map batch: " << to_str(has_map_batch())
-      << "  uprobe refcount (depends on Build:bcc bpf_attach_uprobe refcount): "
-      << to_str(has_uprobe_refcnt()) << std::endl;
-
-  buf << "Map types" << std::endl
-      << "  hash: " << to_str(has_map_hash())
-      << "  array: " << to_str(has_map_array())
-      << "  percpu array: " << to_str(has_map_percpu_array())
-      << "  stack_trace: " << to_str(has_map_stack_trace())
-      << "  perf_event_array: " << to_str(has_map_perf_event_array())
-      << "  ringbuf: " << to_str(has_map_ringbuf()) << std::endl;
-
-  buf << "Probe types" << std::endl
-      << "  kprobe: " << to_str(has_prog_kprobe())
-      << "  tracepoint: " << to_str(has_prog_tracepoint())
-      << "  perf_event: " << to_str(has_prog_perf_event())
-      << "  fentry: " << to_str(has_fentry())
-      << "  kprobe_multi: " << to_str(has_kprobe_multi())
-      << "  uprobe_multi: " << to_str(has_uprobe_multi())
-      << "  iter: " << to_str(has_iter("task")) << std::endl;
+
+  auto to_str = [](bool f) -> std::string { return f ? "yes" : "no"; };
+
+  std::vector<std::pair<std::string, std::string>> helpers = {
+    { "probe_read", to_str(has_helper_probe_read()) },
+    { "probe_read_str", to_str(has_helper_probe_read_str()) },
+    { "probe_read_user", to_str(has_helper_probe_read_user()) },
+    { "probe_read_user_str", to_str(has_helper_probe_read_user_str()) },
+    { "probe_read_kernel", to_str(has_helper_probe_read_kernel()) },
+    { "probe_read_kernel_str", to_str(has_helper_probe_read_kernel_str()) },
+    { "get_current_cgroup_id", to_str(has_helper_get_current_cgroup_id()) },
+    { "send_signal", to_str(has_helper_send_signal()) },
+    { "override_return", to_str(has_helper_override_return()) },
+    { "get_boot_ns", to_str(has_helper_ktime_get_boot_ns()) },
+    { "dpath", to_str(has_d_path()) },
+    { "skboutput", to_str(has_skb_output()) },
+    { "get_tai_ns", to_str(has_helper_ktime_get_tai_ns()) },
+    { "get_func_ip", to_str(has_helper_get_func_ip()) },
+    { "jiffies64", to_str(has_helper_jiffies64()) },
+    { "for_each_map_elem", to_str(has_helper_for_each_map_elem()) }
+  };
+
+  std::vector<std::pair<std::string, std::string>> features = {
+    { "Instruction limit", std::to_string(instruction_limit()) },
+    { "Loop support", to_str(has_loop()) },
+    { "btf", to_str(has_btf()) },
+    { "module btf", to_str(has_module_btf()) },
+    { "Kernel DWARF", to_str(has_kernel_dwarf()) },
+    { "map batch", to_str(has_map_batch()) },
+    { "uprobe refcount", to_str(has_uprobe_refcnt()) }
```

Comment:
Dropping the `(depends on Build:bcc bpf_attach_uprobe refcount)` is probably fine. But perhaps leave it as a comment in the code? The dependency is a little tricky and would be good to keep track of it here.

Reply:
> Dropping the `(depends on Build:bcc bpf_attach_uprobe refcount)` is probably fine. But perhaps leave it as a comment in the code? The dependency is a little tricky and would be good to keep track of it here.
Sure, I'll do that right away!